tests/dhcp_server/test_dhcp_server.py

```diff
@@ -0,0 +1,495 @@
+            return False
+        return True
+    pytest_assert(len([pkt for pkt in pkts if is_expected_pkt(pkt)]) == 0,
+                  "Got unexpected dhcp packet")
+
+
+def validate_dhcp_server_pkts_custom_option(pkts, test_xid, **options):
+    def has_custom_option(pkt):
+        logging.info("validate_dhcp_server_pkts_custom_option: %s" % repr(pkt))
+        if pkt[scapy.BOOTP].xid != test_xid:
+            return False
+        pkt_dhcp_options = pkt[scapy.DHCP].options
+        for option_type, expected_value in options.items():
+            if not match_expected_dhcp_options(pkt_dhcp_options, int(option_type), expected_value):
+                return False
+        return True
+    pytest_assert(len([pkt for pkt in pkts if has_custom_option(pkt)]) == 1,
+                  "Didn't got dhcp packet with expected custom option")
+
+
+def verify_discover_and_request_then_release(
+        duthost,
+        ptfhost,
+        ptfadapter,
+        dut_port_to_capture_pkt,
+        test_xid,
+        ptf_port_index,
+        ptf_mac_port_index,
+        expected_assigned_ip,
+        exp_server_ip,
+        net_mask,
+        refresh_fdb_ptf_port=None
+):
+    client_mac = ptfadapter.dataplane.get_mac(0, ptf_mac_port_index).decode('utf-8')
+    pkts_validator = validate_dhcp_server_pkts if expected_assigned_ip else validate_no_dhcp_server_pkts
+    pkts_validator_args = [test_xid, expected_assigned_ip, DHCP_MESSAGE_TYPE_OFFER_NUM, net_mask, exp_server_ip] \
+        if expected_assigned_ip else [test_xid]
+    discover_pkt = create_dhcp_client_packet(
+        src_mac=client_mac,
+        message_type=DHCP_MESSAGE_TYPE_DISCOVER_NUM,
+        client_options=[],
+        xid=test_xid
+    )
+    send_and_verify(
+        duthost=duthost,
+        ptfhost=ptfhost,
+        ptfadapter=ptfadapter,
+        dut_port_to_capture_pkt=dut_port_to_capture_pkt,
+        ptf_port_index=ptf_port_index,
+        test_pkt=discover_pkt,
+        pkts_validator=pkts_validator,
+        pkts_validator_args=pkts_validator_args,
+        refresh_fdb_ptf_port=refresh_fdb_ptf_port
+    )
+    request_pkt = create_dhcp_client_packet(
+        src_mac=client_mac,
+        message_type=DHCP_MESSAGE_TYPE_REQUEST_NUM,
+        client_options=[("requested_addr", expected_assigned_ip), ("server_id", exp_server_ip)],
+        xid=test_xid
+    )
+    pkts_validator_args = [test_xid, expected_assigned_ip, DHCP_MESSAGE_TYPE_ACK_NUM, net_mask, exp_server_ip] \
+        if expected_assigned_ip else [test_xid]
+    send_and_verify(
+        duthost=duthost,
+        ptfhost=ptfhost,
+        ptfadapter=ptfadapter,
+        dut_port_to_capture_pkt=dut_port_to_capture_pkt,
+        ptf_port_index=ptf_port_index,
+        test_pkt=request_pkt,
+        pkts_validator=pkts_validator,
+        pkts_validator_args=pkts_validator_args,
+        refresh_fdb_ptf_port=refresh_fdb_ptf_port
+    )
+    if expected_assigned_ip:
```

Comment:
check lease by `show lease` before and after release?

Reply:
Yes, added function verify_lease_via_cli